File: state/NewOrderState.java
package state;

public class NewOrderState implements OrderState {
    private final OrderManagement orderManagement;

    public NewOrderState(OrderManagement orderManagement) {
        this.orderManagement = orderManagement;
    }

    @Override
    public void processOrder() {
        System.out.println("Order: " + orderManagement.getOrder().getName() + " is being processing now...");
        orderManagement.changeState(new ProcessingOrderState(orderManagement));
    }

    @Override
    public void shipOrder() {
        System.out.println("Cannot ship the order: " + orderManagement.getOrder().getName()
                + " at the current state...");
    }

    @Override
    public void deliverOrder() {
        System.out.println("Cannot deliver the order: " + orderManagement.getOrder().getName()
                + " at the current state...");
    }

    @Override
    public void cancelOrder() {
        System.out.println("Cancelling order: " + orderManagement.getOrder().getName());
        orderManagement.changeState(new CancelledOrderState(orderManagement));
    }
}

File: PHP/Patterns/src/Behavioral/State/NewOrderState.php
<?php

namespace src\Behavioral\State;

class NewOrderState implements OrderState
{
    private OrderManagement $orderManagement;

    /**
     * @param OrderManagement $orderManagement
     */
    public function __construct(OrderManagement $orderManagement)
    {
        $this->orderManagement = $orderManagement;
    }

    function processOrder()
    {
        print_r("Order: " . $this->orderManagement->getOrder()->getName() . " is being processing now... \n");
        $this->orderManagement->changeState(new ProcessingOrderState($this->orderManagement));
    }

    function shipOrder()
    {
        print_r("Can't ship the order: " .$this->orderManagement->getOrder()->getName(). " at the current state... \n");
    }

    function deliverOrder()
    {
        print_r("Can't deliver the order: " .$this->orderManagement->getOrder()->getName(). " at the current state... \n");
    }

    function cancelOrder()
    {
        print_r('Order: ' . $this->orderManagement->getOrder()->getName() . " is being cancelling now... \n");
        $this->orderManagement->changeState(new CancelledOrderState($this->orderManagement));
    }
}